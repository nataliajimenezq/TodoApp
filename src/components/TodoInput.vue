<template>
  <input
    type="text"
    class="new-todo"
    autofocus
    :placeholder="placehoder"
    :value="modelValue"
    @input="$emit('update:modelValue', $event.target.value)"
    @keyup.enter="$emit('addTodo', modelValue)"
  />
</template>
<script>
export default {
  name: 'TodoInput',

  props: {
    modelValue: {
      type: [String, Number],
    },
    placehoder: {
      type: [String],
      default: 'Whats needs to be doneï¼Ÿ',
    },
  },
};
</script>
