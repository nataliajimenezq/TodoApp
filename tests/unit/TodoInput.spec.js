import { shallowMount } from '@vue/test-utils';
import TodoInput from '@/components/TodoInput.vue';

describe('TodoInput.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TodoInput, {
      props: {
        modelValue: '',
      },
    });
  });

  test('Render component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('fills input with target value ', async () => {
    const inputTag = wrapper.find('input');
    await inputTag.setValue('Go to Gym');
    expect(wrapper.find('input').element.value).toBe('Go to Gym');
  });

  test('fills input with target value ', async () => {
    const inputTag = wrapper.find('input');
    const message = 'Go to Gym';
    await inputTag.setValue(message);
    await inputTag.trigger('keyup.enter');
    expect(wrapper.emitted('addTodo', message));
  });
});
