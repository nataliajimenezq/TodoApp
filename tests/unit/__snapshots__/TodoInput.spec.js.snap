// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput.vue Render component 1`] = `<input type="text" class="new-todo" autofocus="" placeholder="Whats needs to be doneï¼Ÿ">`;
